### ISSUE TITLE
Prevent slow loris attacks

### ISSUE THREAD

**Issue by erikzenkerLogmein** [zaphoyd/websocketpp#789]
Hi zaphoyd,

a default configured websocketpp server can be attacked very easily by a slow loris attack. To prevent this a little bit you can set the handshake open timeout. But I think this is not enough.

We would like to work on more mechanism to secure the websocketpp server. Are you willing to accept pull requests regarding this topic?

Best
Erik

**Comment by vinniefalco** [comment]
Do you know if Boost.Beast websockets are vulnerable to this attack? https://github.com/boostorg/beast

**Comment by erikzenkerLogmein** [comment]
In general Boost.Beast is a much lower layer than websocketpp, right? So I guess, if you implement websockets based on Boost.Beast you have to implement a defence strategy by yourself e.g.: set the proper timeouts, restrict max number of sockets per source ip etc. 

Boost.Beast has the advantage that you are able to implement strategies by your own. Websocketpp hides the internals and makes it hard to inject the code.

**Comment by vinniefalco** [comment]
Yes, it is a little bit lower level (not "much more"). The next version has buit-in timeouts:
https://github.com/boostorg/beast/blob/develop/include/boost/beast/websocket/stream_base.hpp#L71

**Comment by zaphoyd** [comment]
@erikzenkerLogmein security related pull requests are certainly welcome. If you have a rough outline of the strategy you want to try I may be able to point you to the easiest areas of the code to make those sorts of changes.

**Comment by erikzenkerLogmein** [comment]
@vinniefalco Thats very cool! We are experimenting with the boost beast. Maybe, even more security related issues can be tackled. I made a list of defence strategies. I am not totally sure if some of them are already implemented and if all of them make sense:

**Comment by vinniefalco** [comment]
please open an issue in the repo with this list so it can be tracked
